packages/tailwindcss/src/utilities.ts

```diff
@@ -1951,99 +1951,127 @@ export function createUtilities(theme: Theme) {
   staticUtility('break-all', [['word-break', 'break-all']])
   staticUtility('break-keep', [['word-break', 'break-keep']])
 
-  // rounded-*
-  functionalUtility('rounded', {
-    themeKeys: ['--radius'],
-    handle: (value) => [decl('border-radius', value)],
-  })
+  {
+    function handleBareBorderRadiusValue({ value }: NamedUtilityValue) {
+      switch (value) {
+        case 'none':
+          return '0px'
+        case 'full':
+          return 'calc(infinity * 1px)'
+        default:
+          return null
+      }
+    }
 
-  functionalUtility('rounded-s', {
-    themeKeys: ['--radius'],
-    handle: (value) => [
-      decl('border-start-start-radius', value),
-      decl('border-end-start-radius', value),
-    ],
-  })
+    // rounded-*
+    functionalUtility('rounded', {
+      themeKeys: ['--radius'],
+      handleBareValue: handleBareBorderRadiusValue,
```

Comment:
These should probably be implemented using `staticUtility` instead of `handleBareValue`.

Reply:
hmm yeah I'm not sure — it just felt like it should be staticUtility to me _because_ we do know the list of declarations statically.

For filter I actually think `filter-none` should be a static utility too. The fact that it isn't seems like an oversight.